Missing overload `xt::unique(xt::view(...))`

I was trying to

```cpp
#include <xtensor/xarray.hpp>
#include <xtensor/xview.hpp>
#include <xtensor/xsort.hpp>
#include <xtensor/xio.hpp>

int main()
{
  xt::xarray<size_t> a = xt::arange<size_t>(5*5).reshape({5,5});
  auto b = xt::view(a, xt::range(0,3), xt::range(0,3));
  auto c = xt::unique(b);
  return 0;
}
```

but this gives the following compiler error:

```bash
In file included from test.cpp:3:
/usr/local/include/xtensor/xsort.hpp:136:15: error: no matching constructor for initialization of 'xt::xview<xt::xarray_container<xt::uvector<unsigned long, std::__1::allocator<unsigned long> >,
      xt::layout_type::row_major, xt::svector<unsigned long, 4, std::__1::allocator<unsigned long>, true>, xt::xtensor_expression_tag> &, xt::xrange<long>, xt::xrange<long> >'
            R ev;
              ^
/usr/local/include/xtensor/xsort.hpp:149:24: note: in instantiation of function template specialization 'xt::detail::flat_sort_impl<xt::xview<xt::xarray_container<xt::uvector<unsigned long,
      std::__1::allocator<unsigned long> >, xt::layout_type::row_major, xt::svector<unsigned long, 4, std::__1::allocator<unsigned long>, true>, xt::xtensor_expression_tag> &, xt::xrange<long>, xt::xrange<long>
      >, xt::xview<xt::xarray_container<xt::uvector<unsigned long, std::__1::allocator<unsigned long> >, xt::layout_type::row_major, xt::svector<unsigned long, 4, std::__1::allocator<unsigned long>, true>,
      xt::xtensor_expression_tag> &, xt::xrange<long>, xt::xrange<long> > >' requested here
        return detail::flat_sort_impl(e);
                       ^
/usr/local/include/xtensor/xsort.hpp:924:23: note: in instantiation of function template specialization 'xt::sort<xt::xview<xt::xarray_container<xt::uvector<unsigned long, std::__1::allocator<unsigned long> >,
      xt::layout_type::row_major, xt::svector<unsigned long, 4, std::__1::allocator<unsigned long>, true>, xt::xtensor_expression_tag> &, xt::xrange<long>, xt::xrange<long> > >' requested here
        auto sorted = sort(e, xnone());
                      ^
test.cpp:10:16: note: in instantiation of function template specialization 'xt::unique<xt::xview<xt::xarray_container<xt::uvector<unsigned long, std::__1::allocator<unsigned long> >, xt::layout_type::row_major,
      xt::svector<unsigned long, 4, std::__1::allocator<unsigned long>, true>, xt::xtensor_expression_tag> &, xt::xrange<long>, xt::xrange<long> > >' requested here
  auto c = xt::unique(b);
               ^
/usr/local/include/xtensor/xview.hpp:411:18: note: candidate constructor template not viable: requires at least 2 arguments, but 0 were provided
        explicit xview(CTA&& e, FSL&& first_slice, SL&&... slices) noexcept;
                 ^
/usr/local/include/xtensor/xview.hpp:617:18: note: candidate constructor template not viable: requires at least 3 arguments, but 0 were provided
        explicit xview(std::true_type, CTA&& e, FSL&& first_slice, SL&&... slices) noexcept;
                 ^
/usr/local/include/xtensor/xview.hpp:620:18: note: candidate constructor template not viable: requires at least 3 arguments, but 0 were provided
        explicit xview(std::false_type, CTA&& e, FSL&& first_slice, SL&&... slices) noexcept;
                 ^
/usr/local/include/xtensor/xview.hpp:413:9: note: candidate constructor not viable: requires 1 argument, but 0 were provided
        xview(const xview&) = default;
        ^
1 error generated.
```

--- a/include/xtensor/xsort.hpp
+++ b/include/xtensor/xsort.hpp
@@ -111,25 +111,33 @@ namespace xt
         }
 
         template <class VT>
-        struct flatten_sort_result_type
+        struct flatten_sort_result_type_impl
         {
             using type = VT;
         };
 
         template <class VT, std::size_t N, layout_type L>
-        struct flatten_sort_result_type<xtensor<VT, N, L>>
+        struct flatten_sort_result_type_impl<xtensor<VT, N, L>>
         {
             using type = xtensor<VT, 1, L>;
         };
 
         template <class VT, class S, layout_type L>
-        struct flatten_sort_result_type<xtensor_fixed<VT, S, L>>
+        struct flatten_sort_result_type_impl<xtensor_fixed<VT, S, L>>
         {
             using type = xtensor_fixed<VT, xshape<fixed_compute_size<S>::value>, L>;
         };
 
+        template <class VT>
+        struct flatten_sort_result_type
+            : flatten_sort_result_type_impl<typename VT::temporary_type>
+        {
+        };
 
-        template <class E, class R = typename flatten_sort_result_type<E>::type>
+        template <class VT>
+        using flatten_sort_result_type_t = typename flatten_sort_result_type<VT>::type;
+
+        template <class E, class R = flatten_sort_result_type_t<E>>
         inline auto flat_sort_impl(const xexpression<E>& e)
         {
             const auto& de = e.derived_cast();
@@ -384,7 +392,7 @@ namespace xt
      *
      * @return partially sorted xcontainer
      */
-    template <class E, class C, class R = typename detail::flatten_sort_result_type<E>::type,
+    template <class E, class C, class R = detail::flatten_sort_result_type_t<E>,
               class = std::enable_if_t<!std::is_integral<C>::value, int>>
     inline R partition(const xexpression<E>& e, const C& kth_container, placeholders::xtuph /*ax*/)
     {
@@ -411,20 +419,20 @@ namespace xt
     }
 
 #ifdef X_OLD_CLANG
-    template <class E, class I, class R = typename detail::flatten_sort_result_type<E>::type>
+    template <class E, class I, class R = detail::flatten_sort_result_type_t<E>>
     inline R partition(const xexpression<E>& e, std::initializer_list<I> kth_container, placeholders::xtuph tag)
     {
         return partition(e, xtl::forward_sequence<std::vector<std::size_t>, decltype(kth_container)>(kth_container), tag);
     }
 #else
-    template <class E, class I, std::size_t N, class R = typename detail::flatten_sort_result_type<E>::type>
+    template <class E, class I, std::size_t N, class R = detail::flatten_sort_result_type_t<E>>
     inline R partition(const xexpression<E>& e, const I(&kth_container)[N], placeholders::xtuph tag)
     {
         return partition(e, xtl::forward_sequence<std::array<std::size_t, N>, decltype(kth_container)>(kth_container), tag);
     }
 #endif
 
-    template <class E, class R = typename detail::flatten_sort_result_type<E>::type>
+    template <class E, class R = detail::flatten_sort_result_type_t<E>>
     inline R partition(const xexpression<E>& e, std::size_t kth, placeholders::xtuph tag)
     {
         return partition(e, std::array<std::size_t, 1>({kth}), tag);

--- a/test/test_xsort.cpp
+++ b/test/test_xsort.cpp
@@ -216,6 +216,10 @@ namespace xt
         xarray<double> bb = {{1,2,3}, {7,8,9}, {4,5,6}, {7,8,9}};
         xarray<double> bbx = {1,2,3,4,5,6,7,8,9};
         EXPECT_EQ(unique(bb), bbx);
+
+        auto c = view(b, range(0, 3), range(0, 3));
+        auto d = unique(c);
+        EXPECT_EQ(d, bx);
     }
 
     TEST(xsort, setdiff1d)

